scalding-core/src/main/scala/com/twitter/scalding/Source.scala

```diff
@@ -51,7 +51,10 @@ class InvalidSourceException(message: String) extends RuntimeException(message)
  *
  * hdfsPaths represents user-supplied list that was detected as not containing any valid paths.
  */
-class InvalidSourceTap(val hdfsPaths: Iterable[String]) extends SourceTap[JobConf, RecordReader[_, _]] {
+class InvalidSourceTap(val e: Throwable) extends SourceTap[JobConf, RecordReader[_, _]] {
+
+  def this(hdfsPaths: Iterable[String]) =
+    this(new InvalidSourceException(s"No good paths in $hdfsPaths"))
```

Comment:
this is a 2ndary constructor, so there is no `e` on line 56/57, we are creating `e` and passing it to the primary constructor.

So hdfsPaths comes from the caller, eg:

```
new InvalidSourceTap(somePaths)
```

This is for backwards compatibility as previously InvalidSourceTap's constructor took a list of paths. Does that make sense?